openmc/model/surface_composite.py

```diff
@@ -1079,3 +1079,75 @@ def offset(self, distance):
         disp_vec = distance / costheta * unit_nvec
 
         return type(self)(self.points + disp_vec, basis=self.basis)
+
+
+class CruciformPrism(CompositeSurface):
+    """Generalized cruciform prism
+
+    This surface represents a prism parallel to an axis formed by planes at
+    multiple distances from the center. Equivalent to the 'gcross' derived
+    surface in Serpent.
+
+    .. versionadded:: 0.13.4
+
+    Parameters
+    ----------
+    distances : iterable of float
+        A monotonically increasing (or decreasing) iterable of distances in [cm]
+        that form the planes of the generalized cruciform.
+    center : iterable of float
+        The center of the prism in the two non-parallel axes (e.g., (x, y) when
+        axis is 'z') in [cm]
+    axis : {'x', 'y', 'z'}
+        Axis to which the prism is parallel
+    **kwargs
+        Keyword arguments passed to underlying plane classes
+
+    """
+
+    def __init__(self, distances, center=(0., 0.), axis='z', **kwargs):
+        x0, y0 = center
+        self.distances = distances
```

Comment:
This is great! Could we make `distances` a property with a setter that disallows non-positive values and ensures the values are monotonically increasing or decreasing? I might have a tiny preference for only allowing monotonically increasing, but if you want to keep both options go for it.

Reply:
:heavy_check_mark: